diff --git a/meson.build b/meson.build
index 2143401..daf1a62 100644
--- a/meson.build
+++ b/meson.build
@@ -126,7 +126,7 @@ create_ca_test = executable(
         dependencies: [],
     install:false,
 )
-test('create_ca_test', create_ca_test)
+test('create_ca_test', create_ca_test, timeout: 120)
 
 create_csr_test = executable(
     'create_csr_test',
@@ -144,7 +144,7 @@ generate_rsa_key_test = executable(
         dependencies: [],
     install:false,
 )
-test('generate_rsa_key_test', generate_rsa_key_test)
+test('generate_rsa_key_test', generate_rsa_key_test, timeout: 120)
 
 generate_serial_test = executable(
     'generate_serial_test',
@@ -176,7 +176,7 @@ dhparams_test = executable(
 # generators. 4096-bit and larger takes over ten minutes, so it's excluded from
 # the test suite by default.
 prime_lengths = [ 512, 1024 ]
-dhparam_timeout = 120
+dhparam_timeout = 240
 
 if get_option('run_slow_tests')
     prime_lengths = prime_lengths + [ 2048, 4096 ]
